fop/fom.c

```diff
@@ -532,7 +533,8 @@ static void thr_addb2_leave(struct m0_loc_thread *thr,
 M0_INTERNAL void m0_fom_wakeup(struct m0_fom *fom)
 {
 	fom->fo_cb.fc_ast.sa_cb = readyit;
-	m0_sm_ast_post(&fom->fo_loc->fl_group, &fom->fo_cb.fc_ast);
+	if (fom->fo_cb.fc_ast.sa_next == NULL)
+		m0_sm_ast_post(&fom->fo_loc->fl_group, &fom->fo_cb.fc_ast);
```

Comment:
And the same problem for other use of "m0_sm_ast_post()" throughout the Motr code.

Reply:
After this patch, there will be only one place left with such a check: https://github.com/Seagate/cortx-motr/blob/fea2d4bf9eb8fa398db48f11f579038d408cb673/cm/pump.c#L473-L474

But there is an additional check for the fom state in the `complete_wakeup()` callback, that's why I didn't change it.

Reply:
I've added lock protecting the check for sa_next and the call to ast_post(), so that wakeup() can be called concurrently also.

Reply:
> After this patch, there will be only one place left with such a check...

I've cleaned up even that one last place.